feat(due): implement due date for tasks

chipanyanwu opened the pull request:
- Implement with `@mui/x-date-pickers/DateTimePicker`.
  - Can select a date with time. able to select time back before today.
  - Will turn red background if past and incomplete.

- Add new field `due` for `task`, default null.

### Feature Preview
![image](https://github.com/user-attachments/assets/c9bf5cfc-2e26-460e-90f4-cac48f646702)

chipanyanwu committed:
implement due date for tasks

chipanyanwu committed:
cleanup

ZL-Asica commented:
I will take a look tmr morning.

ZL-Asica commented:
Just checked the preview. Since this is a mobile app, we should keep the add button for users to click instead of using enter on keyboard only.

![screen](https://github.com/user-attachments/assets/fd241ea0-1008-4f73-95d5-fbe34eed0ad0)

chipanyanwu committed:
bring back add button

chipanyanwu committed:
make due date tag smaller

chipanyanwu committed:
make more responsive

ZL-Asica committed:
refactor(due): Refine codes

- Move DateTimePickers to Home folder.
- Revert theme.js since is same as default values.
- Refactor DateTimePickers to improve consistency.
- Simplify logic in Task, MicroGoal and AddItem components.

ZL-Asica committed:
feat(Task): Set different color for overdue and incomplete tasks
